### PR TITLE
eve/filetypes: eve filetypes are not always plugins; API docs; simpler - v3

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/10591

Includes: https://github.com/OISF/suricata/pull/10159

Other changes:
- Also adds a change to simplify the setup between threaded and non-threaded mode; as this is a breaking change I added an upgrade section to the docs.  But I'm happy with the API now, this bit always bugged me.
- Constify some arguments per https://github.com/OISF/suricata/pull/10591#discussion_r1517399811
- Document, in code for Doxygen the SCEveFileType.

Tickets:
- https://redmine.openinfosecfoundation.org/issues/6838
- https://redmine.openinfosecfoundation.org/issues/6408

### CODE REVIEW

File: src/util-logopenfile.c [hunk 1]
```diff
@@ -828,10 +833,12 @@ static bool LogFileNewThreadedCtx(LogFileCtx *parent_ctx, const char *log_path,
...
         OutputRegisterFileRotationFlag(&thread->rotation_flag);
-    } else if (parent_ctx->type == LOGFILE_TYPE_PLUGIN) {
+    } else if (parent_ctx->type == LOGFILE_TYPE_FILETYPE) {
         entry->slot_number = SC_ATOMIC_ADD(eve_file_id, 1);
-        thread->plugin.plugin->ThreadInit(
-                thread->plugin.init_data, entry->slot_number, &thread->plugin.thread_data);
+        SCLogDebug("%s - thread %d [slot %d]", log_path, entry->internal_thread_id,
+                entry->slot_number);
+        thread->filetype.filetype->ThreadInit(thread->filetype.init_data, entry->internal_thread_id,
+                &thread->filetype.thread_data);
```
Comment (jasonish) [inline src/util-logopenfile.c:841]: @jlucovsky Can you review this change around the threaded stuff..  So the filetype no longer has to worry about it - other than being thread safe of course.

Comment (jlucovsky) [reply src/util-logopenfile.c:841]: Will do. The plugin/filetype should never be concerned with threading outside of its own implementation-specific threading.

Comment (jasonish) [reply src/util-logopenfile.c:841]: > Will do. The plugin/filetype should never be concerned with threading outside of its own implementation-specific threading.

without this pr they all have to, with this pr a plugin that is written for threaded mode will also run in non-threaded mode which isn't true in current git master.

**Comment by jasonish** [comment]
Replaced by https://github.com/OISF/suricata/pull/10636

**Pull request closed without merging**